build.sh

```diff
@@ -624,19 +652,20 @@ function generateBacklinks {
   perl .post-process-partial-backlink-generator.pl "$HTML_TEMP/wattsi-output/index-$1" > "$2/index.html";
 }
 
-# Starts the syntax-highlighting Python server
+# Starts the syntax-highlighting Python server, when appropriate
 # Arguments: none
-# Output:
+# Output: if the server is necessary, then
 # - A server will be running in the background, at $HIGHLIGHT_SERVER_URL
 # - $HIGHLIGHT_SERVER_PID will be set for later use by stopHighlightServer
 function startHighlightServer {
-  HIGHLIGHT_SERVER_ARGS=()
-  $QUIET && HIGHLIGHT_SERVER_ARGS+=( --quiet )
-  # shellcheck disable=SC2068
-  bs-highlighter-server ${HIGHLIGHT_SERVER_ARGS[@]+"${HIGHLIGHT_SERVER_ARGS[@]}"} &
-  HIGHLIGHT_SERVER_PID=$!
+  if [[ "$LOCAL_WATTSI" == "true" ]]; then
+    HIGHLIGHT_SERVER_ARGS=()
+    $QUIET && HIGHLIGHT_SERVER_ARGS+=( --quiet )
+    bs-highlighter-server "${HIGHLIGHT_SERVER_ARGS[@]}" &
```

Comment:
```suggestion
    bs-highlighter-server ${HIGHLIGHT_SERVER_ARGS[@]+"${HIGHLIGHT_SERVER_ARGS[@]}"} &
```
For non-quiet builds, without using the `${HIGHLIGHT_SERVER_ARGS[@]+"${HIGHLIGHT_SERVER_ARGS[@]}"` expression, the build fails in my environment (and I would think in anybody else’s too…) — because we’re using `set -o nounset` (aka `set -u`), and so we get the following error :

```
./build.sh: line 665: HIGHLIGHT_SERVER_ARGS[@]: unbound variable
```

`${HIGHLIGHT_SERVER_ARGS[@]+"${HIGHLIGHT_SERVER_ARGS[@]}"}` means “Expand the `HIGHLIGHT_SERVER_ARGS[@]` variable only if it’s not unset”.

`"${HIGHLIGHT_SERVER_ARGS[@]}"` by itself means “Unconditionally try to expand the `HIGHLIGHT_SERVER_ARGS[@]` variable, even if it’s unset.

If we weren’t using `set -o nounset`, then `bs-highlighter-server "${HIGHLIGHT_SERVER_ARGS[@]}"` would work fine too. But because we are using `set -o nounset`, then if `HIGHLIGHT_SERVER_ARGS[@]` is unset, it fails.

Reply:
Very interesting. It did not fail on my environment (both Windows Git Bash and Debian 10 via Windows Subsystem for Linux). I guess the idea is that empty arrays count as "unset"?

Why do we not need to do a similar thing for `UNZIP_ARGS` on line 638?

Reply:
Maybe it's because this was fixed in recent (2016+) Bash versions? https://stackoverflow.com/a/39687362/3191

Reply:
> Maybe it's because this was fixed in recent (2016+) Bash versions? https://stackoverflow.com/a/39687362/3191

Yeah, probably so. I know the bash that ships in macOS is quite old.

Reply:
> Very interesting. It did not fail on my environment (both Windows Git Bash and Debian 10 via Windows Subsystem for Linux). I guess the idea is that empty arrays count as "unset"?

Yes, to sh and bash they do at least.

> Why do we not need to do a similar thing for `UNZIP_ARGS` on line 638?

I think we do need to do a similar thing with that too. I guess the only reason none of us has run into it causing a failure is that none of us has run the build script with the verbose switch since whenever it was that `set -o nounset` was added to the script

Reply:
Sounds good. I'll push a separate commit that adds that, although I'll want to rearrange and squash it before merging.

Reply:
OK, actually I ended up force-pushing, because I realized I had messed up the commit message for the third commit anyway. PTAL.

Reply:
> > Why do we not need to do a similar thing for `UNZIP_ARGS` on line 638?
> 
> I think we do need to do a similar thing with that too.

Actually, I think I misspoke about that. Line 639 is what matters, because that’s where `UNZIP_ARGS` is actually referenced. And when it’s referenced there, it’s always going to have been assigned/set — because line 638 adds `"$HTML_TEMP/wattsi-output.zip" -d "$2"` to its value unconditionally.

Reply:
Ah right, that makes sense! Force-pushed to undo that line (and remove the discussion of it from the commit message).

Reply:
> Ah right, that makes sense! Force-pushed to undo that line (and remove the discussion of it from the commit message).

ah, OK — re-pulled now and will review and test it again to be sure